openstack/containerorchestration/v1/common/errors.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	"strings"
+)
+
+// ErrorResponse represents failed response
+type ErrorResponse struct {
+	gophercloud.ErrUnexpectedResponseCode
+	Errors []Error `json:"errors" required:"true"`
+}
+
+// Error represents a Magnum API error message
+type Error struct {
+	Status    int    `json:"status"`
+	Code      string `json:"code"`
+	Title     string `json:"title"`
+	Detail    string `json:"detail"`
+	RequestID string `json:"request_id,omitempty"`
+}
+
+// Error returns the error message details from the server
+func (e *ErrorResponse) Error() string {
+	errors := []string{}
+	for _, e := range e.Errors {
+		errors = append(errors, e.Detail)
+	}
+	return strings.Join(errors, "\n")
+}
+
+func (e *ErrorResponse) unwrapError(r gophercloud.ErrUnexpectedResponseCode) bool {
+	e.ErrUnexpectedResponseCode = r
+	err := json.Unmarshal(r.Body, &e)
+	return err == nil
+}
+
+// Error400 extracts the actual error message from the body of the response
+func (e *ErrorResponse) Error400(r gophercloud.ErrUnexpectedResponseCode) error {
+	if e.unwrapError(r) {
+		return e
+	} else {
```

Comment:
Because of the `return` in the `if` clause, we can get rid of all these `else`s.